agent/run_job.go

```diff
@@ -428,6 +439,24 @@ func (r *JobRunner) cleanup(ctx context.Context, wg *sync.WaitGroup, exit core.P
 	r.agentLogger.Info("Finished job %s for build at %s", r.conf.Job.ID, r.conf.Job.Env["BUILDKITE_BUILD_URL"])
 }
 
+// waitForKubernetesProcessesToComplete waits for Kubernetes command containers to finish
+// before stopping log collection, ensuring post-command hook logs are captured.
+func (r *JobRunner) waitForKubernetesProcessesToComplete(ctx context.Context) {
+	r.agentLogger.Debug("[JobRunner] Waiting for Kubernetes processes to complete before stopping log collection")
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, defaultKubernetesLogCollectionGracePeriod)
+	defer cancel()
+
+	select {
+	case <-r.process.Done():
+		r.agentLogger.Debug("[JobRunner] Kubernetes processes completed, stopping log collection")
+	case <-timeoutCtx.Done():
+		r.agentLogger.Info("[JobRunner] Timeout waiting for Kubernetes processes, stopping log collection")
+	}
+
+	time.Sleep(1 * time.Second)
```

Comment:
What's this for?

Reply:
@DrJosh9000 most probably unnecessary, or could be replaced with a better mechanism to allow k8s handle job cleanup gracefully.